homeassistant/components/cover/opengarage.py

```diff
@@ -0,0 +1,246 @@
+"""
+Platform for the opengarage.io cover component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/cover.opengarage
+"""
+import logging
+
+import voluptuous as vol
+
+import requests
+
+from homeassistant.components.cover import (
+    CoverDevice, PLATFORM_SCHEMA, SUPPORT_OPEN, SUPPORT_CLOSE)
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.const import (
+    CONF_DEVICE, CONF_NAME, STATE_UNKNOWN, STATE_CLOSED, STATE_OPEN,
+    CONF_COVERS, CONF_HOST, CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+DEFAULT_NAME = 'OpenGarage'
+DEFAULT_PORT = 80
+
+CONF_DEVICEKEY = "device_key"
+
+ATTR_SIGNAL_STRENGTH = "wifi_signal"
+ATTR_DISTANCE_SENSOR = "distance_sensor"
+ATTR_DOOR_STATE = "door_state"
+
+STATE_OPENING = "opening"
+STATE_CLOSING = "closing"
+STATE_STOPPED = "stopped"
+STATE_OFFLINE = "offline"
+
+STATES_MAP = {
+    0: STATE_CLOSED,
+    1: STATE_OPEN
+}
+
+
+# Validation of the user's configuration
+COVER_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Required(CONF_DEVICEKEY): cv.string,
+    vol.Optional(CONF_NAME): cv.string
```

Comment:
`vol.Required` should go first then `vol.Optional`, preferably alphabetically ordered. Make it easier to keep the documentation in sync.